src/legacy/core_plugins/kibana/public/discover/np_ready/components/sidebar/discover_field_details.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { EuiLink, EuiSpacer, EuiIconTip, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { DiscoverFieldBucket } from './discover_field_bucket';
+import { getWarnings } from './lib/get_warnings';
+import { Bucket, FieldDetails } from './types';
+import { IndexPatternField, IndexPattern } from '../../../../../../../../plugins/data/public';
+
+interface DiscoverFieldDetailsProps {
+  field: IndexPatternField;
+  indexPattern: IndexPattern;
+  details: FieldDetails;
+  onAddFilter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
+}
+
+export function DiscoverFieldDetails({
+  field,
+  indexPattern,
+  details,
+  onAddFilter,
+}: DiscoverFieldDetailsProps) {
+  const warnings = getWarnings(field);
+  if (!details) {
```

Comment:
If details could be potentially falsy, these cases should be presented in types I think.
`details: FieldDetails | null | undefined` or whatever you check.

Reply:
maybe this was the case a time ago, but now details can's be falsy. thx for paying attention to such details!

Reply:
change that